### DISCUSSION TITLE
Deriving Serialize/Deserialize on Payload to persist to hand-editable files.

### DISCUSSION THREAD

**Discussion by omrischwarz** [ahlner/sparkplug-rs#6] [Ideas]
For testing purposes, I'd like to be able to record messages and persist them in a hand editable format for testing purposes. 
Is there any reason not to just add #derive[(Serialize...)] to the structure definitions?

**Comment by ahlner** [comment]
Serde is no longer supported in rust-protobuf 3 but there is an alternative approach: https://docs.rs/protobuf-json-mapping/latest/protobuf_json_mapping/

Can you please try to use this? Feedback is very welcome :)
